lab_1_classify_by_unigrams/main.py

```diff
@@ -5,50 +5,81 @@
 """
 # pylint:disable=too-many-locals, unused-argument, unused-variable
 
-
 def tokenize(text: str) -> list[str] | None:
     """
     Split a text into tokens.
 
-    Convert the tokens into lowercase, remove punctuation, digits and other symbols
+    Convert the tokens into lowercase, remove punctuation, digits, and other symbols.
 
     Args:
-        text (str): A text
+        text (str): A text.
 
     Returns:
-        list[str] | None: A list of lower-cased tokens without punctuation
+        list[str] | None: A list of lower-cased tokens without punctuation.
 
-    In case of corrupt input arguments, None is returned
+    In case of corrupt input arguments, None is returned.
     """
+    if not isinstance(text, str):
+        return []
+
+    tokens = []
+
+    for token in text:
+        if token.isalpha():
+            tokens.append(token.lower())
+
+    return tokens
 
 
 def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
     """
     Calculate frequencies of given tokens.
 
     Args:
-        tokens (list[str] | None): A list of tokens
+        tokens (list[str] | None): A list of tokens.
 
     Returns:
-        dict[str, float] | None: A dictionary with frequencies
+        dict[str, float] | None: A dictionary with frequencies.
 
-    In case of corrupt input arguments, None is returned
+    In case of corrupt input arguments, None is returned.
     """
+    if not isinstance(tokens, list):
+        return {}
+
+    frequency_dict = {}
+
+    for letter in tokens:
+        if letter.isalpha():
+            if letter in frequency_dict:
+                frequency_dict[letter] += 1
+            else:
+                frequency_dict[letter] = 1
```

Comment:
the final result will be the same: in any case, it starts with a counter that will be equal to 1 for the first occurrence of the letter, since zero plus one gives one